source/_drafts/Feature-Spotlight-Advanced-Plugins.md

```diff
@@ -0,0 +1,203 @@
+
+These plugins - which we'll call _basic plugins_ - are lightweight and offer complete control of the player. That's really useful and **it isn't changing** - existing plugins should continue to work!
+
+But what if you want a richer set of features? Or more guidance on how to structure your plugin? Or more tools out of the box that help manage complex plugin-rich players?
+
+Well, until Video.js 6.0, you had to figure things out on your own.
+
+## Introducing Advanced Plugins
+
+One of Video.js' strengths is its rich ecosystem of plugins; so, in the last few months, we wanted to focus our efforts on improving the plugin author experience.
+
+While projects like [the plugin generator][generator] and [videojs-spellbook][spellbook] make becoming a plugin author easier than ever, the Video.js team thought it was important to provide a foundational API and set of conventions on which the future of Video.js plugins could be built.
+
+Our solution is _advanced plugins_.
+
+### Advanced Plugins are Component-Like
+
+One of the design goals for advanced plugins was to provide an API that was reminiscent of the existing components system. We achieved this in a number of ways. 
+
+At the lowest level, this included a name change for the plugin registration function from `videojs.plugin` to `videojs.registerPlugin` (taking a naming cue from `videojs.registerComponent` and `videojs.registerTech`).
+
+Beyond a simple registration method name change, advanced plugins are _class-based_. A trivial example of an advanced plugin might look something like this:
+
+```js
+const Plugin = videojs.getPlugin('Plugin');
+
+class HelloWorld extends Plugin {
+  constructor(player) {
+    super(player);
+    this.addClass('hello-world')
+  }
+}
+
+videojs.registerPlugin('helloWorld', HelloWorld);
+```
+
+This plugin can be initialized in the same way as a basic plugin - via a player method whose name matches the registered name of the plugin.
+
+In the case of advanced plugins, this method is a factory function, which instantiates the plugin class and returns an instance.
+
+It's useful to know that the player method that is created will _always_ be a function. If a player already has an instance of an advanced plugin, its associated method will simply return the pre-existing instance rather than re-initialize it:
+
+```js
+const player = videojs('my-player');
+const instance = player.helloWorld();
+
+// Logs: 'true'
+videojs.log(instance === player.helloWorld());
+```
+
+The `helloWorld` method will return this plugin object until it is disposed - after which it will create a new plugin instance again.
+
+### Events
+
+Similar to components, advanced plugins can listen to and trigger events via the `on`, `one`, `off`, and `trigger` methods.
+
+This provides a loosely coupled communication channel for plugins and other objects (components, players, etc) to manage their own state and respond to changes in the state of one another.
+
+#### Plugin Event Hash
```

Comment:
Is this something we used in the videojs docs? The term seems new and unfamiliar. Could be substitute `hash` with something simpler that reaches a wider audience?